dependencies.yaml

```diff
@@ -184,6 +184,7 @@ dependencies:
         packages:
           - numba>=0.57
           - numpy>=1.21
+          - dunamai
```

Comment:
We may want to make this a dependency only in development environments since it's being edited out of all packages (both wheels and conda).

Reply:
For now I've removed runtime version detection, so dunamai is no longer a dependency of anything except the build process itself.